File: go/c-vs-go/calc.go
package main

import (

	"fmt"
	"time"
)

func main() {
	x := 0.0
	start := time.Now()
	for i := 1.0; i < 100000000; i++ {
		tmp := 10000+i*0.01
		x +=  tmp*tmp - tmp
	}
	consume := time.Since(start)
	fmt.Printf("calc %e in go, took %s\n", x, consume)
}

File: go/c-vs-go/calc.c
#include<stdio.h>
#include<sys/time.h>
long long  get_current_ms() {
    struct timeval t;
    gettimeofday(&t,NULL);
    return t.tv_sec * 1000 + t.tv_usec / 1000;
}

int main() {
    long long start_ms = get_current_ms(); 
    double sum = 0.0;
    for (int i = 0; i < 100000000; i ++) {
        double tmp = 10000 + 0.01 * i;
        sum += tmp * tmp - tmp;
    }
    long long end_ms = get_current_ms();
    printf("calc %e in c, cost %lld ms\n", sum, end_ms - start_ms);
}
